src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -588,10 +588,6 @@ private void DoStart()
                 }
                 if (logger.IsVerbose) { logger.Verbose("Message gateway service started successfully."); }
 
-                scheduler.QueueTask(clientRegistrar.Start, clientRegistrar.SchedulingContext)
-                    .WaitWithThrow(initTimeout);
```

Comment:
I'm struggling a little to understand how this could be throwing a null ref exception, since the old code was calling `ClientObserverRegistrar.Start` only after the gateway was set (assuming he didn't replace any internals so that the cast operation in line 584 didn't fail). Of course since we didn't have the full stacktrace it's really hard to know exactly where it was throwing, but did you find a path where `ClientObserverRegistrar.gateway` could have been null?